File: ArduinoCore/source/main/src/Bt/Com/BinaryOutputPackage.cpp
//*************************************************************************************************
//
//  BITTAILOR.CH - ArduinoCore
//
//-------------------------------------------------------------------------------------------------
//
//  Bt::Com::BinaryOutputPackage
//  
//*************************************************************************************************

#include "Bt/Com/BinaryOutputPackage.hpp"

namespace Bt {
namespace Com {


//-------------------------------------------------------------------------------------------------

BinaryOutputPackage::BinaryOutputPackage(I_PackageBuffer& iBuffer)
: mBuffer(&iBuffer){

}

//-------------------------------------------------------------------------------------------------

BinaryOutputPackage::~BinaryOutputPackage() {

}
//-------------------------------------------------------------------------------------------------

I_PackageBuffer& BinaryOutputPackage::buffer() {
   return *mBuffer;
}
//-------------------------------------------------------------------------------------------------

I_OutputPackage& BinaryOutputPackage::write(bool iValue) {
   uint8_t value = iValue ? 1 : 0;
   mBuffer->put(value);
   return *this;
}

//-------------------------------------------------------------------------------------------------

I_OutputPackage& BinaryOutputPackage::write(int8_t iValue) {
   mBuffer->put(static_cast<uint8_t>(iValue));
   return *this;
}

//-------------------------------------------------------------------------------------------------

I_OutputPackage& BinaryOutputPackage::write(uint8_t iValue) {
   mBuffer->put(iValue);
   return *this;
}

//-------------------------------------------------------------------------------------------------

} // namespace Com
} // namespace Bt

File: JRgbScreenClient/src/ch/bittailor/bt/com/BinaryOutputPackage.java
package ch.bittailor.bt.com;

public class BinaryOutputPackage implements IOutputPackage {

	private IPackageBuffer mBuffer;
	
	public BinaryOutputPackage(IPackageBuffer iBuffer) {
		this.mBuffer = iBuffer;
	}

	@Override
	public IOutputPackage writeBoolean(boolean iValue) {
		if (iValue) {
			mBuffer.put((byte)1);
		} else {
			mBuffer.put((byte)0);
		}
		return this;
	}

	@Override
	public IOutputPackage writeInt8(int iValue) {
		mBuffer.put((byte)iValue);
		return this;
	}

	@Override
	public IOutputPackage writeUInt8(int iValue) {
		mBuffer.put((byte)iValue);
		return this;
	}

}
